src/boot/StoreProvider.js

```diff
@@ -11,11 +15,38 @@ type Props = $ReadOnly<{|
 |}>;
 
 export default class StoreProvider extends PureComponent<Props> {
+  unsubscribeStoreObserver: () => void;
+
   componentDidMount() {
     timing.start('Store hydration');
     restore(() => {
       timing.end('Store hydration');
     });
+
+    this.unsubscribeStoreObserver = observeStore(
+      store,
+      // onChange will fire when this value changes
+      state => tryGetActiveAccount(state)?.zulipVersion,
+      (prevZulipVersion, zulipVersion) => {
+        configureScope(scope => {
+          // Set server version tags on Sentry's global scope, so
+          // all events will have them.
+          //
+          // If `zulipVersion` is falsy, all values in the object
+          // passed to `setTags` will be undefined. This means the
+          // tags will be removed from the scope, though this is
+          // unofficial as a way to remove tags:
+          //   https://github.com/getsentry/sentry-javascript/pull/3108#issue-534072956
+          scope.setTags(getServerVersionTags(zulipVersion));
+        });
```

Comment:
This part is all pretty specific to the nuances of the Sentry API. So let's have it live in the code whose job naturally includes knowing about Sentry, like `src/utils/logging.js`.

As a bonus, then it can stay next to `getServerVersionTags`. The call that this StoreProvider code would make could then look like
```js
  logging.setTagsFromServerVersion(zulipVersion);
```